libsplinter/src/network/connection_manager/mod.rs

```diff
@@ -350,11 +350,11 @@ fn handle_request(req: CmRequest, state: &mut ConnectionState) {
 }
 
 fn notify_subscribers(
-    subscribers: &mut HashMap<String, SyncSender<Vec<CmNotification>>>,
-    notifications: Vec<CmNotification>,
+    subscribers: &mut HashMap<String, SyncSender<CmNotification>>,
+    notification: CmNotification,
 ) {
     for (id, sender) in subscribers.clone() {
```

Comment:
Couldn't say which is more efficient here: using `clone`, and remove if it fails or using `retain`, like we have in other situations.  Probably the later, since it only creates a new vector, and doesn't clone the contents.